### ISSUE TITLE
Create a pipeline that builds JupyterBook HTML

### ISSUE THREAD

**Issue by tumido** [AICoE/meteor#15]
As a Data Scientist using Python,
I want to use Meteor to turn a repo full of Jupyter Notebooks into a JupyterBook container image,
So that the container image gets deployed to Operate First,
And So that I can show off my work.

:warning: EDIT :boom: :bomb: 
**Initial implementation**

Create a Tekton pipeline that can be co-located (co-hosted) with AICoE CI.

1. This pipeline is triggered via a webhook and consumes a git url and a UUID which is passed from the web interface
2. It runs `jupyter-book build` as shown [here](https://jupyterbook.org/start/build.html#build-your-books-html), which results in HTML files being produced
3. Use the HTML as an input to s2i-nginx
4. deliver container image to quay

**Don't consider discussion the bellow for now**
The discussion below expands beyond this initial design and is more of a brainstorming that we'll may implement in the future in next iterations of Meteor. 

Possible expansions in the future:

1. Convert any repo to a basic JupyterBook format before rendering HTML
2. ~Store data in Object bucket instead of a PVC~

**Comment by tumido** [comment]
@oindrillac sure we can do that...


What about we have a script in this pipeline that:
- will use `README.md` as the landing page
- will locate all `.md` files and writes them to the `_toc.yaml` *

That way we don't need to convert repos to the jupyterbook format and just use them as is

\* something like:

```sh
$ cd ocp-ci-analysis
$ find . -not -path "*/\.*" -name "*.md" -exec sh -c "echo '- file: {}' | cut -f 1,2 -d '.'" \;
- file: ./docs/aiml-cicd-market-research
- file: ./docs/cell_labels_in_testgrid
- file: ./manifests/README
- file: ./notebooks/README
- file: ./notebooks/failure-type-classification/README
- file: ./README
```

**Comment by oindrillac** [comment]
@tumido yes, exactly what I was thinking. That would make sense 👍

**Comment by MichaelClifford** [comment]
This would be great @tumido we also want to include the `.ipynb` files into the book as well. not just the `.md`

**Comment by MichaelClifford** [comment]
Does this also make #13 redundant?

**Comment by tumido** [comment]
> Does this also make #13 redundant?

That is probably correct. Considering that repositories that are input to meteor can be of various formats we may offer 2 options for convenience:

- if a repo is already in the JupyterBook format (contains both `_toc.yaml` and `_config.yaml` ) we use it as is
- if a repo is not in JupyterBook we may use the script above to convert it.

If we apply the above - for the first iteration of Project Meteor it is probably easier just to convert the Elyra tutorials to JupyterBook manually than creating and debugging a "JypyterBook generator" pipeline. WDYT? I think we may want to keep the JypyterBook pipeline simple in first iteration and extend it with the above later on.

**Comment by MichaelClifford** [comment]
I like it. So we will do it manually for the first iteration (#13 remains valid) and use a simpler pipeline that expects a repo in the correct format. :+1: We can automate the Jupyterbook set up later.

**Comment by tumido** [comment]
AI: @tumido @harshad16 document this task better in here.

**Comment by pacospace** [comment]
- It should not consider .github templates already present
- It should not place docs repo in docs/docs

cc @MichaelClifford

**Comment by durandom** [comment]
Instead of editing the original issue:

1. the MVP would be an s2i image that builds a book, adds the outup to the image and serves it via a webserver. It's probably an extension of the available nginx/httpd s2i images

that probably means getting rid of the UUID and PVC bucket.
Maybe create a new issue for the s2i image and then revisit this issue?

**Comment by goern** [comment]
@tumido please edit 1.: where does the UUID come from?

**Comment by pacospace** [comment]
Depends-On: https://github.com/AICoE/meteor/issues/48

**Comment by tumido** [comment]
> @tumido please edit 1.: where does the UUID come from?

@goern  thanks, updated. It comes from the web interface and is generated when the "order form" is submitted. It's meant to couple the order with images and order detail page and what not.

**Comment by tumido** [comment]
Resolved via: https://github.com/AICoE/meteor-operator/pull/6
